### PR TITLE
[bug] Fix setting arguments exceeding the max arg num

### PR DESCRIPTION
**Pull request by lin-hitonami**
Before this PR, if the number of arguments exceeds the maximum number of arguments, it still sets the argument to the array beyond the index. 

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #7622
* #7612
* __->__ #7623
* #7611
* #7610
* #7608
* #7576

### CODE REVIEW

File: python/taichi/lang/kernel_impl.py [hunk 1]
```diff
@@ -655,10 +655,15 @@ def func__(*args):
 
             actual_argument_slot = 0
             launch_ctx = t_kernel.make_launch_context()
```
Comment (bobcao3) [inline python/taichi/lang/kernel_impl.py:657]: I think this check should be done within launch context...

Comment (lin-hitonami) [reply python/taichi/lang/kernel_impl.py:657]: In fact this should be done when compiling the kernel. Let's keep this as a temporary solution. I'll refactor it later.

Comment (lin-hitonami) [reply python/taichi/lang/kernel_impl.py:657]: BTW the limit will be removed as soon as the refactoring completes.

**Review by ailzhang — APPROVED** [review]
Yea the whole section below can be refactored/simplified in compilation, looking forward to it!

**Pull request merged**